src/main/kotlin/com/hedvig/underwriter/util/Pii.kt

```diff
@@ -0,0 +1,39 @@
+package com.hedvig.underwriter.util
+
+import java.lang.reflect.Modifier
+import java.util.LinkedList
+
+@Target(AnnotationTarget.FIELD)
+@Retention(AnnotationRetention.RUNTIME)
+annotation class Pii()
+
+fun Any?.toNonPiiString(): String =
+    if (this == null) "null" else reflectionToString(this)
```

Comment:
Should this one maybe check for its conformance to `Map` and `Collection` first?

```suggestion
fun Any?.toNonPiiString(): String = when (this) {
  null -> "null"
  is Map<*, *> -> this.toNonPiiString()
  is Collection<*> -> this.toNonPiiString()
  else -> reflectionToString(this)
```

Reply:
I think otherwise that which implementation is run is decided at compile time, since the two implementations below aren't actually overrides.

Reply:
Good point. But as you say, I do not think we need to care about performance optimisations here so let postpone that.

Reply:
Hmm - this is not for performance though. It's rather that the wrong method will be called:
```kotlin
val map = mapOf("key", "value")
map.toNonPiiString() // this will call the  `Map<*, *>` version
val anyMap: Any = map
anyMap.toNonPiiString() // this will call the `Any` version - which will miss the custom formatting for Maps
```

Reply:
I think this is also important due to this part below: `prop.get(obj)?.toNonPiiString()`. That will be called on an `Any`, which could make it lose the information about which implementation it should use:
```kotlin
data class PeopleDatabase(
  val data: Map<String, Person>
)

val database = PeopleDatabase(mapOf("id" to Person("Fredrik Tõnisson-Bystam"))
// the initial call will take the right path, but the subequent call to `prop.get(obj)?.toNonPiiString()`
// on the `data` field will end up at the `.startsWith("java") and thus print the Person without its
// fullName masked
database.toNonPiiString()
```

Reply:
Haha, I got this mixed up.

Did not see this, good point, will look into it!

Reply:
Fixed together with Elvins re-naming in new commit.

Reply:
Awesome! 🎉